src/Microsoft.IdentityModel.Tokens/InMemoryCryptoProviderCache.cs

```diff
@@ -118,22 +158,40 @@ public override bool TryAdd(SignatureProvider signatureProvider)
                 return false;
 
             var cacheKey = GetCacheKey(signatureProvider);
+#if NET461 || NET472 || NETSTANDARD2_0 
+            MemoryCache signatureProviderCache;
+#elif NET45
+            MaxCapacityRefreshCache<string, SignatureProvider> signatureProviderCache;
+#endif
+            // Determine if we are caching a signing or a verifying SignatureProvider.
             if (signatureProvider.WillCreateSignatures)
+                signatureProviderCache = _signingSignatureProviders;
+            else
+                signatureProviderCache = _verifyingSignatureProviders;
+
+#if NET461 || NET472 || NETSTANDARD2_0 
+            // The cache does NOT already have a crypto provider associated with this key.
+            if (!signatureProviderCache.TryGetValue(cacheKey, out _))
             {
-                if (_signingSignatureProviders.TryAdd(cacheKey, signatureProvider))
+                var cacheEntryOptions = new MemoryCacheEntryOptions
```

Comment:
we could inline this 'new' in the call to .Set and remove the local variable.

Reply:
Good point; I've removed the local variable and inlined it.